contracts/citycoin-logic-v1.clar

```diff
@@ -1,967 +1,80 @@
-    (get-entitled-stacking-reward stacker target-reward-cycle block-height)
-)
-
-;; Determine how many uSTX a Stacker is allowed to claim, given the reward cycle they Stacked in and the current block height.
-;; This method only returns a positive value if:
-;; * The current block height is in a subsequent reward cycle
-;; * The Stacker actually did lock up some tokens in the target reward cycle
-;; * The Stacker locked up _enough_ tokens to get at least one uSTX.
-;; It's possible to Stack tokens but not receive uSTX.  For example, no miners may have mined in this reward cycle.
-;; As another example, you may have Stacked so few that you'd be entitled to less than 1 uSTX.
-(define-private (get-entitled-stacking-reward (stacker principal) (target-reward-cycle uint) (cur-block-height uint))
-    (let (
-        (stacked-this-cycle
-            (get amount-token
-                (default-to { amount-token: u0 }
-                    (map-get? stacked-per-cycle { stacker: stacker, reward-cycle: target-reward-cycle }))))
-        (total-tokens-this-cycle
-            (default-to { total-ustx: u0, total-tokens: u0 }
-                (map-get? tokens-per-cycle { reward-cycle: target-reward-cycle })))
-    )
-    (match (get-reward-cycle cur-block-height)
-        cur-reward-cycle
-          (if (or (<= cur-reward-cycle target-reward-cycle) (is-eq u0 (get total-tokens total-tokens-this-cycle)))
-              ;; either this reward cycle hasn't finished yet, or the Stacker contributed nothing
-              u0
-              ;; (total-ustx * this-stackers-tokens) / total-tokens-stacked
-              (/ (* (get total-ustx total-tokens-this-cycle) stacked-this-cycle) 
-                 (get total-tokens total-tokens-this-cycle))
-          )
-        ;; before first reward cycle
-        u0
-    ))
-)
-
-;; Helper variable used only to achieve "dynamic" filter condition in closure function called by `filter`
-(define-data-var miner-to-kick (optional {miner-id: uint, amount-ustx: uint}) none)
-
-;; Inner filter function.
-;; Returns true if supplied miner in not miner who committed the least and needs to be kicked out of the list
-(define-private (remove-miner-to-kick-closure (miner-data {miner-id: uint, amount-ustx: uint}))
-    (not 
-        (is-eq miner-data (unwrap-panic (var-get miner-to-kick)))
-    )
+(define-private (get-reward-cycle (stacksHeight uint))
+  (default-to u0 (contract-call? .citycoin-core get-reward-cycle stacksHeight))
 )
 
-(define-private (find-least (miner-data {miner-id: uint, amount-ustx: uint}) 
-                            (ret-data (optional {miner-id: uint, amount-ustx: uint})))
-    (if (is-some ret-data)
-        (if ( < (get amount-ustx miner-data) (default-to u0 (get amount-ustx ret-data)))
-            (some miner-data)
-            ret-data
-        )
-        (some miner-data)
-    )
+(define-private (stacking-active-at-cycle (rewardCycle uint))
+  (contract-call? .citycoin-core stacking-active-at-cycle rewardCycle)
```

Comment:
No, one thing to clarify here is this would be a set of contracts per city, not one `citycoin-core` to oversee them all. Once this is complete as a template, everything will be renamed to:

- `miamicoin-core`
- `miamicoin-logic-v1`
- `miamicoin-token`
- etc...

I think I need to outline this clearer in the documentation somewhere :slightly_smiling_face: